src/main/java/com/shipmentEvents/demo/SQS.java

```diff
@@ -0,0 +1,134 @@
+ * AWS SDK for Java.
+ * <p>
+ * <b>Prerequisites:</b> You must have a valid Amazon Web
+ * Services developer account, and be signed up to use Amazon SQS. For more
+ * information on Amazon SQS, see http://aws.amazon.com/sqs.
+ * <p>
+ * Fill in your AWS access credentials in the provided credentials file
+ * template, and be sure to move the file to the default location
+ * (~/.aws/credentials) where the sample code will load the credentials from.
+ * <p>
+ * <b>WARNING:</b> To avoid accidental leakage of your credentials, DO NOT keep
+ * the credentials file in your source directory.
+ */
+public class SQS {
+
+    public static void main(String[] args) throws Exception {
+
+        /*
+         * The ProfileCredentialsProvider will return your [default]
+         * credential profile by reading from the credentials file located at
+         * (~/.aws/credentials).
+         */
+        ProfileCredentialsProvider credentialsProvider = new ProfileCredentialsProvider();
+        try {
+            credentialsProvider.getCredentials();
+        } catch (Exception e) {
+            throw new AmazonClientException(
+                    "Cannot load the credentials from the credential profiles file. " +
+                            "Please make sure that your credentials file is at the correct " +
+                            "location (~/.aws/credentials), and is in valid format.",
+                    e);
+        }
+
+        AmazonSQS sqs = AmazonSQSClientBuilder.standard()
+                .withCredentials(credentialsProvider)
+                .withRegion(Regions.US_WEST_2)
+                .build();
+
+        System.out.println("===========================================");
+        System.out.println("Getting Started with Amazon SQS");
+        System.out.println("===========================================\n");
+
+        try {
+            // Create a queue
+            System.out.println("Creating a new SQS queue called MyQueue.\n");
+            CreateQueueRequest createQueueRequest = new CreateQueueRequest("MyQueue");
+            String myQueueUrl = sqs.createQueue(createQueueRequest).getQueueUrl();
+
+            // List queues
+            System.out.println("Listing all queues in your account.\n");
+            for (String queueUrl : sqs.listQueues().getQueueUrls()) {
+                System.out.println("  QueueUrl: " + queueUrl);
+            }
+            System.out.println();
+
+            // Send a message
+            System.out.println("Sending a message to MyQueue.\n");
+            sqs.sendMessage(new SendMessageRequest(myQueueUrl, "This is my message text."));
+
+            // Receive messages
+            System.out.println("Receiving messages from MyQueue.\n");
+            ReceiveMessageRequest receiveMessageRequest = new ReceiveMessageRequest(myQueueUrl);
+            List<Message> messages = sqs.receiveMessage(receiveMessageRequest).getMessages();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Enable long polling to reduce the number of empty responses when no messages are available in reply to a ReceiveMessage request and to eliminate false empty responses. This can reduce your cost of using Amazon SQS. To enable long polling, set the wait time in seconds on the ReceiveMessageRequest object supplied to the receiveMessage method or set the wait time in seconds on the ReceiveMessageWaitTimeSeconds attribute of the AmazonSQS class.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-sqs-long-polling.html)